### PR TITLE
docs: add verification instructions in Hardhat example

### LINKED ISSUE #2054: Include verification setup in Hardhat example
### Problem

The Hardhat example does not provide instructions nor setup to verify the contract. It would be useful to devs to have an end-to-end solution to build, test, deploy and **verify** their smart contracts on Hedera.

### Solution

Include verification instructions in the Hardhat example.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by acuarica**
Fixes 
- #2054.

> The Hardhat example does not provide instructions nor setup to verify the contract. It would be useful to devs to have an end-to-end solution to build, test, deploy and verify their smart contracts on Hedera.

**Description**:

This PR adds instructions to setup the verification of deployed smart contracts in a custom Sourcify instance. To do so, it includes a new `env` key to set up the user private key on _testnet_ and updates Hardhat dependencies. More specifically, `ethers.js`

There are breaking changes when migrating `ethers.js` _v5_ to _v6_ (this is because we had to update `hardhat-toolbox` for the custom verification to work).

- `getBalance` needs to be invoked from the `provider`
- Use `target` instead of `address` to get the address of a contract
- Use `METHOD.staticCall` instead of `callStatic.METHOD`
- Use `getCreateAddress` instead of `getContractAddress`
- Use `logs` instead of `events` to get the emitted events in a receipt
- Use native BigInt instead of BigNumber

Some of these changes can be found in the official `ethers.js` docs https://docs.ethers.org/v6/migrating/.

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: tools/hardhat-example/.env.example [hunk 1]
```diff
@@ -4,3 +4,5 @@ RECEIVER_PRIVATE_KEY=0x2e1d968b041d84dd120a5860cee60cd83f9374ef527ca86996317ada3
 RELAY_ENDPOINT='http://localhost:7546'
 
 MNEMONIC="here is where your twelve words mnemonic should be put my friend"
+
+TESTNET_OPERATOR_PRIVATE_KEY='<if you want to deploy on testnet go to https://portal.hedera.com/ to setup your private key>'
```
Comment (AlfredoG87) [inline tools/hardhat-example/.env.example:8]: I like this 👍

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Comment by AlfredoG87** [comment]
Looks like there is an issue with the key, it is expecting 32 bytes:

>  Invalid account: #0 for network: testnet - private key too long, expected 32 bytes

File: tools/hardhat-example/.env.example (hunk 1 above)
Comment (AlfredoG87) [reply tools/hardhat-example/.env.example:8]: Althought I think this is breaking the CI

Comment (Nana-EC) [reply tools/hardhat-example/.env.example:8]: Q: why is this needed vs just changing your `OPERATOR_PRIVATE_KEY`
The current value assumes local node per the comment but if you changed networks it makes sense to change value no?

Comment (acuarica) [reply tools/hardhat-example/.env.example:8]: Yes and no. We could indeed use the same key. But having different ones allows the user to deploy to different networks at the same time without changing the configuration or the env file.

**Comment by acuarica** [comment]
Hardhat example CI job has been fixed and all other CI jobs now pass, pls take a look.

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**